cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.save;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.logging.LoggingContext;
+import co.cask.cdap.logging.LoggingConfiguration;
+import co.cask.cdap.logging.appender.kafka.LoggingEventSerializer;
+import co.cask.cdap.logging.kafka.KafkaLogEvent;
+import co.cask.cdap.logging.write.AvroFileWriter;
+import co.cask.cdap.logging.write.FileMetaDataManager;
+import co.cask.cdap.logging.write.LogFileWriter;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.RowSortedTable;
+import com.google.common.collect.TreeBasedTable;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.inject.Inject;
+import org.apache.twill.common.Threads;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Plugin that writes the log data.
+ */
+public class KafkaLogWriterPlugin extends AbstractIdleService implements KafkaLogProcessor {
```

Comment:
If `KafkaLogWriterPlugin` extends `AbstractIdleService`, then we can call `stop` instead of `cleanUp`, right?